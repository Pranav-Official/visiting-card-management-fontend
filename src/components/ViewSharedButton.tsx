import React, { ReactNode } from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import colors from '../utils/colorPallete';
import SharePersonIcon from '..//assets/images/shareContactIcon.svg';
interface ViewSharedButtonProps {
  children?: ReactNode;
  title: string;
  number: string;
  onPressing: () => any;
  highlighted?: boolean;
}
const ViewSharedButton: React.FC<ViewSharedButtonProps> = ({
  children,
  title,
  number,
  onPressing,
  highlighted = false,
}) => {
  return (
    <TouchableOpacity onPress={() => onPressing()} style={styles.mainContainer}>
        <View style={styles.lightBlueContainer}></View>
        <View style={styles.contentContainer}>
          <Text style={styles.titleText}>{title}</Text>
          <View style={styles.contactNumberContainer}>
            <SharePersonIcon width={45} height={45} />
            <Text style={styles.contactText}>{number} contacts</Text>
          </View>
        </View>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  mainContainer: {
    flexDirection: 'row',
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 5,
  },
  lightBlueContainer: {
    backgroundColor: colors['primary-accent'],
    width: '7%',
  },
  contentContainer: {
    // flex:1,
    padding: 20,
  },
  contactNumberContainer: {
    flexDirection: 'row',
    gap: 20,
  },
  contactText: {
    flex: 1,
    fontSize: 28,
    padding: 5,
  },
  titleText: {
    fontSize: 32,
    color: 'black',
  },
});
export default ViewSharedButton;
