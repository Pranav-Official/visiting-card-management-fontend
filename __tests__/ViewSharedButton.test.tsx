import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ViewSharedButton from '../src/components/ViewSharedButton';

describe('ViewSharedButton Component', () => {
  const mockTitle = 'Test Title';
  const mockNumber = '5';
  const mockOnPressing = jest.fn();

  it('renders title correctly', () => {
    const { getByText } = render(
      <ViewSharedButton
        title={mockTitle}
        number={mockNumber}
        onPressing={mockOnPressing}
      />,
    );
    expect(getByText(mockTitle)).toBeTruthy();
  });

  it('renders number correctly', () => {
    const { getByText } = render(
      <ViewSharedButton
        title={mockTitle}
        number={mockNumber}
        onPressing={mockOnPressing}
      />,
    );
    expect(getByText(`${mockNumber} contacts`)).toBeTruthy();
  });

  it('calls onPressing prop when pressed', () => {
    const { getByTestId } = render(
      <ViewSharedButton
        title={mockTitle}
        number={mockNumber}
        onPressing={mockOnPressing}
      />,
    );
    const button = getByTestId('sharedBtntest');
    fireEvent.press(button);
    expect(mockOnPressing).toHaveBeenCalled();
  });

  
});
