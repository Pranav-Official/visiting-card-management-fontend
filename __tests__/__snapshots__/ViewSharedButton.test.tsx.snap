// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSharedButton Component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#A1D5E3",
        "width": "7%",
      }
    }
  />
  <View
    style={
      {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 32,
        }
      }
    >
      Test Title
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 20,
        }
      }
    >
      <shareContactIcon.svg
        height={45}
        width={45}
      />
      <Text
        style={
          {
            "flex": 1,
            "fontSize": 28,
            "padding": 5,
          }
        }
      >
        5
         contacts
      </Text>
    </View>
  </View>
</View>
`;
